spec/lib/synapse/haproxy_spec.rb

```diff
@@ -21,7 +21,18 @@ class MockWatcher; end;
   let(:mockwatcher_with_server_options) do
     mockWatcher = double(Synapse::ServiceWatcher)
     allow(mockWatcher).to receive(:name).and_return('example_service2')
-    backends = [{ 'host' => 'somehost', 'port' => 5555, 'haproxy_server_options' => 'backup'}]
+    backends = [{ 'host' => 'somehost', 'port' => 5555, 'haproxy_server_options' => 'id 12 backup'}]
```

Comment:
I refactored the haproxy_server_id test to test all three ways of specifying ids (no specification, one with haproxy_server_id and one with haproxy_server_options). Test failed before I changed the algorithm for picking ids (duplicate id of 1) and then passed after.